dapps/shop/backend/routes/auth.js

```diff
@@ -1,7 +1,190 @@
+const omit = require('lodash/omit')
+const { Sellers, Shops } = require('../data/db')
+const {
+  createSalt,
+  hashPassword,
+  AuthSeller,
+  authenticatedAsSeller
+} = require('./_combinedAuth')
+const { IS_PROD } = require('../utils/const')
+const encConf = require('../utils/encryptedConfig')
+const { validateConfig, validateShop } = require('../utils/validators')
 
 module.exports = function(app) {
-  app.get('/auth', auth, (req, res) => {
+  app.get('/auth', (req, res) => {
+    res.json({ success: req.isAuthenticated() })
+  })
+
+  app.get('/auth/:email', async (req, res) => {
+    // TODO: Add some rate limiting here
+    const seller = await Sellers.findOne({
+      where: {
+        email: req.params.email
+      }
+    })
+    return res.sendStatus(seller === null ? 404 : 204)
+  })
+
+  app.post(
+    '/auth/login',
+    (req, res, next) => {
+      return next()
+    },
+    AuthSeller,
+    (req, res) => {
+      res.json({ success: req.isAuthenticated() })
+    }
+  )
+
+  // TODO: Should this at least use API key auth?
+  app.post('/auth/registration', async (req, res) => {
+    const { name, email, password } = req.body
+    if (!name || !email || !password) {
+      return res.status(400).json({
+        success: false,
+        message: 'Invalid registration'
+      })
+    }
+
+    const sellerCheck = await Sellers.findOne({
+      where: {
+        email
+      }
+    })
+
+    if (sellerCheck) {
+      return res.status(409).json({
+        success: false,
+        message: 'Registration exists'
+      })
+    }
+
+    const salt = await createSalt()
+    const passwordHash = await hashPassword(salt, password)
+
+    const seller = await Sellers.create({
+      name,
+      email,
+      password: passwordHash
+    })
+
+    if (!seller) {
+      return res.json({ success: false })
+    }
+
+    return res.json({ success: true })
+  })
+
+  app.delete('/auth/registration', authenticatedAsSeller, async (req, res) => {
+    const { id } = req.user
+
+    if (!id) {
+      return res.status(400).json({ success: false })
+    }
+
+    const destroy = await Sellers.destroy({
+      where: {
+        id
+      }
+    })
+
+    res.json({ success: false, destroy })
+  })
+
+  app.get('/shop', authenticatedAsSeller, async (req, res) => {
+    const { id } = req.user
+
+    if (!id) {
+      return res.status(400).json({ success: false })
+    }
+
+    const rows = await Shops.findAll({
+      where: {
+        seller_id: id
+      }
+    })
+
+    const shops = rows.map(row => {
+      return omit(row.dataValues, ['config', 'seller_id'])
+    })
+
+    res.json({
+      success: true,
+      shops
+    })
+  })
+
+  app.post('/shop', authenticatedAsSeller, async (req, res) => {
+    const shopObj = req.body
+
+    if (!validateShop(shopObj)) {
+      return res
+        .status(400)
+        .json({ success: false, message: 'Invalid shop data' })
+    }
+
+    const shop = await Shops.create({
+      ...shopObj,
```

Comment:
I think the model is still expecting underscored, and the API is expecting camelcased in the validation. I'm not sure if part of the problem here is sqlite.

The way I've done this in other APIs is define everything in JavaScript as camelcased (models, attributes, the works), set `underscored: true`, and let Sequelize translate between JavaScript and SQL. It'll use underscored column names.

Reply:
Will get this straightened out.  Gonna take a minute.  Basically everything that interacts with thes emodels needs to change.

Reply:
This has been updated and deployed.  This shouldn't require any DB upgrades or anything so past versions should be compat, but the ORM interaction will be all camelCase.  Still need to go through the dapp admin real quick but this hold be resolved.